assets/tests/components/detours/diversionPage.test.tsx

```diff
@@ -242,6 +242,70 @@ describe("DiversionPage", () => {
     expect(screen.queryByTitle("Detour End")).toBeNull()
   })
 
+  test("shows 'Regular Route' text when the detour is finished", async () => {
+    jest.mocked(fetchDetourDirections).mockResolvedValue(
+      ok(
+        detourShapeFactory.build({
+          directions: [
+            { instruction: "Turn left on Main Street" },
+            { instruction: "Turn right on High Street" },
+            { instruction: "Turn sharp right on Broadway" },
+          ],
+        })
+      )
+    )
+
+    jest
+      .mocked(fetchFinishedDetour)
+      .mockResolvedValue(finishedDetourFactory.build())
+
+    const { container } = render(<DiversionPage />)
+
+    act(() => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+
+    act(() => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+
+    expect(await screen.findByText("Regular Route")).toBeVisible()
+  })
+
+  test("does not show 'Regular Route' when detour is not finished", async () => {
+    jest.mocked(fetchDetourDirections).mockResolvedValue(
+      ok(
+        detourShapeFactory.build({
+          directions: [
+            { instruction: "Turn left on Main Street" },
+            { instruction: "Turn right on High Street" },
+            { instruction: "Turn sharp right on Broadway" },
+          ],
+        })
+      )
+    )
+
+    jest
+      .mocked(fetchFinishedDetour)
+      .mockResolvedValue(finishedDetourFactory.build())
+
+    const { container } = render(<DiversionPage />)
+
+    expect(screen.queryByText("Regular Route")).not.toBeInTheDocument()
+
+    act(() => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+
+    act(() => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+
+    fireEvent.click(screen.getByRole("button", { name: "Undo" }))
```

Comment:
nit(non-blocking): hmm, these seem to be undoing each-other, and we're not waiting on the results of the `Undo` click, could we remove both of these?

```suggestion
```

Reply:
Oh derp, embarrassing. My thought process was that I want to verify that the "regular route" message is absent in both situations where a detour is "unfinished" -- pre-completion of detour, and post-undo. I can tuck a `expect(await screen.findByText("Regular Route")).toBeVisible()` before the undo, but then we have repeated tests, essentially

Reply:
It _sort of_ feels like this is actually two tests hiding in one. First, testing that `Regular Route` doesn't appear before connecting a detour, and second, testing that `Regular Route` disappears after undoing a connected detour. 

Please treat ☝️ as musings, not things to act on right now, unless you want to :). When we come to a clearer testing strategy overall, that'll be a good opportunity to rework some of these tests.

Reply:
It's true, I did want to lump two tests together, heh. they are "similar vibe" but different things 😅